<launch>
    <node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="screen" args="--delete_db_on_start">
        <param name="frame_id" type="string" value="base_link"/>
        <param name="subscribe_depth" type="bool" value="true"/>
        <param name="subscribe_scan"  type="bool" value="true"/>
        <param name="queue_size" type="int" value="10"/>

        <remap from="odom" to="mbot_odom"/>
        <remap from="rgb/image" to="/camera/rgb/image_raw"/>
        <remap from="depth/image" to="/camera/depth_registered/image_raw"/>
        <remap from="rgb/camera_info" to="/camera/rgb/camera_info"/>

        <param name="RGBD/NeighborLinkRefining" value="true"/>  <!-- Do odometry correction with consecutive laser scans -->
        <param name="RGBD/ProximityBySpace"     value="true"/>  <!-- Local loop closure detection (using estimated position) with locations in WM -->
        <param name="RGBD/ProximityByTime"      value="false"/> <!-- Local loop closure detection with locations in STM -->
        <param name="Reg/Strategy"              value="1"/>     <!-- 0=Visual, 1=ICP, 2=Visual+ICP -->
        <param name="Vis/InlierDistance"        value="0.1"/>   <!-- 3D visual words correspondence distance -->
        <param name="RGBD/OptimizeFromGraphEnd" value="false"/> <!-- Optimize graph from initial node so /map -> /odom transform will be generated -->
        <param name="Reg/Force3DoF"             value="true"/>
        <param name="Grid/FromDepth"             value="true"/> <!-- output octomap -->

    </node>
</launch>